closed/src/java.base/share/classes/sun/security/ec/NativeECDSASignature.java

```diff
@@ -102,20 +102,20 @@ abstract class NativeECDSASignature extends SignatureSpi {
     // private key, if initialized for signing
     private ECPrivateKey privateKey;
 
-    // private key impl, if initialized for signing
-    private ECPrivateKeyImpl privateKeyImpl;
+    // native private key pointer, if initialized for signing
+    private long nativePrivateKey;
 
     // public key, if initialized for verifying
     private ECPublicKey publicKey;
 
-    // public key impl, if initialized for verifying
-    private ECPublicKeyImpl publicKeyImpl;
+    // native public key pointer, if initialized for verifying
+    private long nativePublicKey;
 
     // the format (i.e., true for the IEEE P1363 format and false for ASN.1)
     private final boolean p1363Format;
 
     // the Java implementation, if needed
-    private ECDSASignature javaImplementation;
+    protected ECDSASignature javaImplementation;
```

Comment:
Use the minimum viable visibility (package-private) here.

Reply:
I had to change it to allow it to be used by `NativeECDSASignature` subclasses, like `RawECDSA`.

Reply:
Sure, but it doesn't need to be visible outside this package:
```java
    ECDSASignature javaImplementation;
```

Reply:
Oh, ok. Thought you meant to revert to private. Changed.